arch/arm/core/aarch32/cortex_m/mpu/arm_mpu_v7_internal.h

```diff
@@ -216,7 +217,11 @@ static inline int is_in_region(u32_t r_index, u32_t start, u32_t size)
 			MPU_RASR_SIZE_Pos) + 1;
 	r_addr_end = r_addr_start + (1UL << r_size_lshift) - 1;
 
-	if (start >= r_addr_start && (start + size - 1) <= r_addr_end) {
+	if (size == 0 || __builtin_add_overflow(start, size - 1, &end)) {
```

Comment:
`__builtin_add_overflow` may not be available on all compilers (for GCC, you need GCC 5.1 or higher).

Should we add/use a fallback macro instead like Linux kernel does?
https://github.com/torvalds/linux/blob/8b614cb8f1dcac8ca77cf4dd85f46ef3055f8238/include/linux/overflow.h#L55

Reply:
Actually, it turns out we already have something similar:
https://github.com/zephyrproject-rtos/zephyr/blob/4b38659b0661962a3fb9fc98c634944d88236923/include/sys/math_extras_impl.h#L32

Reply:
thanks for pointing it out. Will change it.